c7n/resources/rds.py

```diff
@@ -1819,3 +1821,51 @@ class resource_type(TypeInfo):
         universal_taggable = object()
 
     augment = universal_augment
+
+
+@filters.register('consecutive-snapshots')
+class ConsecutiveSnapshots(Filter):
+    """Returns instances where number of consective daily snapshots is equal to/or greater than n days.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: rds-daily-snapshot-count
+                resource: rds
+                filters:
+                  - type: consecutive-snapshots
+                    days: 7
+    """
+    schema = type_schema('consecutive-snapshots', days={'type': 'number', 'minimum': 1},
+        required=['days'])
+    permissions = ('rds:DescribeDBSnapshots', 'rds:DescribeDBInstances')
+
+    def augment(self, resources):
+        client = local_session(self.manager.session_factory).client('rds')
+
+        for resource in resources:
+            response = client.describe_db_snapshots(
```

Comment:
We should be able to reduce the total number of API calls here by moving the `describe_db_snapshots` call above the loop. A couple considerations though...

* It would require using `Filters` argument rather than `DBInstanceIdentifier`. Looks like the `db-instance-id` filter can handle up to 100 instance IDs per call, so the list would need to be chunked. There's a [convenience function](https://github.com/cloud-custodian/cloud-custodian/blob/b5156031c68fdf7c516c52146ddddc0a10d97b2a/c7n/utils.py#L236-L245) for that, which is used in other [similar cases](https://github.com/cloud-custodian/cloud-custodian/blob/5a17f45b27cc2480917afdde39fcf65c007a71b3/c7n/resources/ec2.py#L863-L882).
* The API can only return 100 snapshots per call, so it would need to be paginated. [This](https://github.com/cloud-custodian/cloud-custodian/blob/31905220950b4ceed4a4cca46febd2da5a1c517a/c7n/resources/route53.py#L245-L247) is a good existing example of fetching a full paginated result set with retries built in.